### PR TITLE
enable transparent draw when having transparent colors - gl-scatter3d module

### PR DESCRIPTION
**Pull request by archmoj**
Please refer to issue https://github.com/plotly/plotly.js/issues/3492 and PR https://github.com/plotly/plotly.js/pull/3496 for more info.
@etpinard

### CODE REVIEW

File: pointcloud.js [hunk 1]
```diff
@@ -606,6 +606,7 @@ proto.update = function(options) {
           for(var j=0; j<4; ++j) {
             color[j] = c[j]
           }
+          if(this.opacity === 1 && c[3] < 1) this.opacity = 0.999
```
Comment (etpinard) [inline pointcloud.js]: Why do need this hacky `0.999` piece?

Comment (archmoj) [reply pointcloud.js]: When `opacity === 1` the `proto.isOpaque` function returns `true` and `proto.isTransparent` function returns `false`. Whereas we need to reverse that.
https://github.com/gl-vis/gl-scatter3d/blob/afc69258bef5a458e83b7d031ccfc6ada8e098fb/pointcloud.js#L118-#L140

Comment (etpinard) [reply pointcloud.js]: Cool. I'm still going to call that _hacky_.

So at the very least, could you add a comment above this line explaining what this does.

Comment (etpinard) [reply pointcloud.js]: That's a much better fix. Thanks!

**Pull request merged**